src/library.js

```diff
@@ -3007,6 +3007,13 @@ LibraryManager.library = {
     var info = _mmap.mappings[start];
     if (!info) return 0;
     if (num == info.num) {
+      // At the Linux man page, it says:
+      // "The file may not actually be updated until msync(2) or munmap(2) are called."
+      // I guess that means we need to call msync when doing munmap
+      _msync(start, num); // todo: which flags?
```

Comment:
Thinking about this. There is PROT_READ, PROT_WRITE, etc. but this can only be implemented when it's available on the underlying file system which is not there on MEMFS. Then there is MAP_PRIVATE which is [described here](http://man7.org/linux/man-pages/man2/mmap.2.html) as:

> Create a private copy-on-write mapping.  Updates to the
> mapping are not visible to other processes mapping the
> same file, and are not carried through to the underlying
> file.  It is unspecified whether changes made to the file
> after the mmap() call are visible in the mapped region.

So we should not propagate the save to filesystem if it's opened like that. MAP_ANONYMOUS and MAP_DENYWRITE are deprecated or ignored. I don't know how to interpret MAP_FIXED.

[This StackOverflow answer](http://stackoverflow.com/questions/5902629/mmap-msync-and-linux-process-termination) says that msync does not have to be forced. I guess not doing the msync only makes sense when opening the map as read-only, but MEMFS does not support that. A backend like NODEFS won't break when doing the msync anyway, ignoring writes anyway. So I think we can just keep it like that. However, we need to implement `MAP_PRIVATE` in MEMFS.